File: python/tools/dashboards/get.py
import json

from schema.tool_inputs import DashboardGetSchema
from tools.types import Context, TextContent, Tool, ToolResult


async def get_dashboard_handler(context: Context, params: DashboardGetSchema) -> ToolResult:
    project_id = await context.get_project_id()
    dashboard_result = await context.api.dashboards(project_id).get(int(params.dashboardId))

    if not dashboard_result.success:
        raise Exception(f"Failed to get dashboard: {dashboard_result.error}")

    return ToolResult(content=[TextContent(text=json.dumps(dashboard_result.data.model_dump()))])


def get_dashboard_tool() -> Tool[DashboardGetSchema]:
    return Tool(
        name="dashboard-get",
        description="Get a specific dashboard by ID.",
        schema=DashboardGetSchema,
        handler=get_dashboard_handler
    )

File: typescript/src/tools/surveys/get.ts
import { SurveyGetSchema } from "@/schema/tool-inputs";
import { formatSurvey } from "@/tools/surveys/utils/survey-utils";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = SurveyGetSchema;
type Params = z.infer<typeof schema>;

export const getHandler = async (context: Context, params: Params) => {
	const { surveyId } = params;
	const projectId = await context.stateManager.getProjectId();

	const surveyResult = await context.api.surveys({ projectId }).get({
		surveyId,
	});

	if (!surveyResult.success) {
		throw new Error(`Failed to get survey: ${surveyResult.error.message}`);
	}

	const formattedSurvey = formatSurvey(surveyResult.data, context, projectId);

	return {
		content: [{ type: "text", text: JSON.stringify(formattedSurvey) }],
	};
};

const definition = getToolDefinition("survey-get");

const tool = (): Tool<typeof schema> => ({
	name: "survey-get",
	description: definition.description,
	schema,
	handler: getHandler,
	annotations: {
		destructiveHint: false,
		idempotentHint: true,
		openWorldHint: true,
		readOnlyHint: true,
	},
});

export default tool;
